sklearn/cross_decomposition/tests/test_pls.py

```diff
@@ -260,6 +261,15 @@ def check_ortho(M, err_msg):
     check_ortho(pls_ca.y_scores_, "y scores are not orthogonal")
 
 
+def test_convergence_fail():
+    d = load_linnerud()
+    X = d.data
+    Y = d.target
+    pls_bynipals = pls_.PLSCanonical(n_components=X.shape[1],
+                                     max_iter=2, tol=1e-10)
+    assert_warns(ConvergenceWarning, pls_bynipals.fit, X, Y)
```

Comment:
As there are multiple sources of convergencewarning here, it would be best to use `assert_warns_message` to check you're triggering the one you expect.

Reply:
As both other `ConvergenceWarning`s were changed back to `UserWarnings` (see outdated comment above) this does not apply anymore, does it?